client/sites-dashboard/utils.ts

```diff
@@ -38,7 +38,10 @@ export const displaySiteUrl = ( siteUrl: string ) => {
 	return siteUrl.replace( 'https://', '' ).replace( 'http://', '' );
 };
 
-export function isCustomDomain( siteSlug: string ): boolean {
+export function isCustomDomain( siteSlug: string | null | undefined ): boolean {
+	if ( ! siteSlug ) {
+		return false;
+	}
 	return ! siteSlug.endsWith( '.wordpress.com' ) && ! siteSlug.endsWith( '.wpcomstaging.com' );
```

Comment:
I thought we could shorten it, but it seems too hacky:

```suggestion
	return !! $siteSlug && ! siteSlug.endsWith( '.wordpress.com' ) && ! siteSlug.endsWith( '.wpcomstaging.com' );
```

Reply:
Thanks for reviewing it.
And thanks for considering and sharing an optimization. I agree that it looks a bit hacky.

I've also considered using this instead:
`! siteSlug?.endsWith( '.wordpress.com' ) && ! siteSlug?.endsWith( '.wpcomstaging.com' );`
But I preferred to be more explicit.